core/trino-parser/src/test/java/io/trino/sql/TestSqlFormatter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.sql.tree.ColumnDefinition;
+import io.trino.sql.tree.CreateTable;
+import io.trino.sql.tree.GenericDataType;
+import io.trino.sql.tree.Identifier;
+import io.trino.sql.tree.NodeLocation;
+import io.trino.sql.tree.QualifiedName;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import static io.trino.sql.SqlFormatter.formatSql;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestSqlFormatter
+{
+    @Test
+    public void testIdentifiers()
+    {
+        // Reserved keyword
+        assertThat(formatSql(new Identifier("exists"))).isEqualTo("\"exists\"");
+        assertThat(formatSql(new Identifier("\"exists\""))).isEqualTo("\"\"\"exists\"\"\"");
+
+        // Non-reserved keyword
+        assertThat(formatSql(new Identifier("analyze"))).isEqualTo("analyze");
+        assertThat(formatSql(new Identifier("\"analyze\""))).isEqualTo("\"\"\"analyze\"\"\"");
+
+        // ANSI-compliant string
+        assertThat(formatSql(new Identifier("account"))).isEqualTo("account");
+        assertThat(formatSql(new Identifier("\"account\""))).isEqualTo("\"\"\"account\"\"\"");
+
+        // Non-ANSI quoted string
+        assertThat(formatSql(new Identifier("\"1\""))).isEqualTo("\"\"\"1\"\"\"");
```

Comment:
I meant testing the same word as quoted and unquoted identifier, like below.
```suggestion
        // Reserved keyword
        assertThat(formatSql(new Identifier("exists", false))).isEqualTo("\"exists\"");
        assertThat(formatSql(new Identifier("exists", true))).isEqualTo("\"exists\"");

        // Non-reserved keyword
        assertThat(formatSql(new Identifier("analyze", false))).isEqualTo("analyze");
        assertThat(formatSql(new Identifier("analyze", true))).isEqualTo("\"analyze\"");

        // ANSI-compliant string
        assertThat(formatSql(new Identifier("account", false))).isEqualTo("account");
        assertThat(formatSql(new Identifier("account", true))).isEqualTo("\"account\"");

        // Non-ANSI quoted string
        assertThat(formatSql(new Identifier("1", true))).isEqualTo("\"1\"");
```

However it is a good idea to test quotes as part of the identifier name. If you want to do this, please add another test method.

Reply:
Done & added another method for quotes as part of identifier name.